File: PledgeVault.Core/Models/Resource.cs
﻿using PledgeVault.Core.Contracts.Entities;
using PledgeVault.Core.Enums.Models;
using System;

namespace PledgeVault.Core.Models;

public sealed class Resource : IEntity
{
    public Resource()
    {
        EntityCreated = DateTime.Now;
        EntityActive = true;
    }

    public int Id { get; set; }

    public DateTime EntityCreated { get; set; }

    public DateTime? EntityModified { get; set; }

    public bool EntityActive { get; set; }

    public string Title { get; set; }

    public string SiteUrl { get; set; }

    public ResourceType ResourceType { get; set; }

    public string Summary { get; set; }

    public int PledgeId { get; init; }

    public Pledge Pledge { get; init; }
}

File: PledgeVault.Web/pledgevault-web/src/models/Resource.ts
import { IResponse } from "../contracts/IResponse";
import ResourceType from "../enums/ResourceType";
import { Pledge } from "./Pledge";

export interface Resource extends IResponse {
  id: number;
  title: string;
  siteUrl: string;
  resourceType: ResourceType;
  summary: string;
  pledgeId: number;
  pledge: Pledge;
}
